docs-ref-services/servicebus.md

```diff
@@ -2,47 +2,200 @@
+Please read [Migration from v0.21.1 to v0.50.0](#migration-from-v0.21.1-to-v0.50.0) to determine if upgrading is
+right for you at this time.
 
-## Install the libraries
+The new AMQP-based API offers improved message passing reliability, performance and expanded feature support going forward.
+
+For documentation on the legacy HTTP-based operations please see [Using HTTP-based operations of the legacy API](#using-http-based-operations-of-the-legacy-api)
+
+
+## Prerequisites
+
+* Azure subscription - [Create a free account](https://azure.microsoft.com/free/)
+* Azure [Service Bus namespace and management credentials](https://docs.microsoft.com/azure/service-bus-messaging/service-bus-create-namespace-portal)
+* [Python 2.7-3.7](https://www.python.org/downloads/)
+
+
+## Installation
 ```bash
 pip install azure-servicebus
 ```
 
-## ServiceBus Queues
-ServiceBus Queues are an alternative to Storage Queues that might be
+## Connect to Azure Service Bus
+
+### Get credentials
+Use the Azure CLI snippet below to populate an environment variable with the Service Bus connection string (you can also find this value in the Azure portal). The snippet is formatted for the Bash shell.
+
+```Bash
+RES_GROUP=<resource-group-name>
+NAMESPACE=<servicebus-namespace>
+
+export SB_CONN_STR=$(az servicebus namespace authorization-rule keys list \
+ --resource-group $RES_GROUP \
+ --namespace-name $NAMESPACE \
+ --name RootManageSharedAccessKey \
+ --query primaryConnectionString \
+ --output tsv)
+```
+
+### Create client
+Once you've populated the `SB_CONN_STR` environment variable, you can create the ServiceBusClient.
+```python
+import os
+from azure.servicebus import ServiceBusClient
+
+connection_str = os.environ['SB_CONN_STR']
+
+sb_client = ServiceBusClient.from_connection_string(connection_str)
+```
+
+
+## Service Bus Queues
+Service Bus Queues are an alternative to Storage Queues that might be
 useful in scenarios where more advanced messaging features are needed
 (larger message sizes, message ordering, single-operation destructive
 reads, scheduled delivery) using push-style delivery (using long
 polling).
 
-The service can use Shared Access Signature authentication, or ACS
-authentication.
+The service can use Shared Access Signature authentication.
+
+### Create queue
+This creates a new queue within the Service Bus namespace. If a queue of the same name already exists an error will be raised. 
+```python
+sb_client.create_queue("MyQueue")
+```
+Optional parameters to configure the queue behaviour can also be specified
+```python
+sb_client.create_queue(
+    "MySessionQueue",
+    requires_session=True  # Create a sessionful queue
+    max_delivery_count=5  # Max delivery attempts per message
+)
+```
+
+### Get a queue client
+A QueueClient can be used to send and receive messages from the queue, along with other operations.
```

Comment:
Ideally, each first reference to a type should be hyperlinked to that type's page in the SDK reference docs.

Reply:
Thanks @mmacy - is there an easy way to know what that URL will be before the ref docs are published? Or should I PR an update to this once the docs go live?

Reply:
@annatisch You can try to infer it if it's not a type already on docs, but that's a bit of a hassle; recommend update this once the ref goes live.